### ISSUE TITLE
Non-top-level functions compiled incorrectly

### ISSUE THREAD

**Issue by pauldraper** [google/closure-compiler#594]
With SIMPLE_OPTMIZATIONS,

``` javascript
if (!Math.PI) {
    function foo() {}
}
foo();
```

compiles to

``` javascript
if(!Math.PI)var foo=function(){};foo();
```

The original runs without error in all major browsers; the second does not.
Plus, it could even be longer than the original (if you had already removed whitespace and braces).
It should output

``` javascript
if(!Math.PI)function foo(){};foo();
```

(You could also move the function declaration to the top of the scope, but probably not with `SIMPLE_OPTIMIZATIONS`, and in `ADVANCED_OPTIMIZATIONS`, `foo` will be inlined anyway.)

Perhaps someone tried to make the output compatible with ES5 strict?

**Comment by bogdanb** [comment]
I get different behavior than reported above with Closure v20190819. Take this code:

    function test() {
        if (globalVar) {
            function scoped(element) {
                console.log("test");
            }
        }
    
        scoped(element);
    }

Compiling with `--language_in=ECMASCRIPT5` generates an error, which seems very reasonable:

> ERROR - [JSC_LANGUAGE_FEATURE] This language feature is only supported for ECMASCRIPT6 mode or better: block-scoped function declaration.

But if I compile with `--language_in=ECMASCRIPT6` (or any newer argument up to ECMASCRIPT_2019) it generates the following code, without any warning or error:

> function test(){resetTriggers(element)};

In other words, the block-scoped function is just eliminated completely. As far as I can understand that’s a bug. It might be arguable what the correct behavior would be (and browsers might disagree), but unconditionally removing the scoped function without any warning or error seems broken.
